tunnel/tunnel.go

```diff
@@ -204,6 +206,30 @@ func (t *gtunnel) SetPcap(fpcap string) error {
 	}
 }
 
+func (t *gtunnel) SetPcapFd(fpcap int32) error {
+	pcap := t.pcapio
+
+	if pcap == nil {
+		return errStackMissing
+	}
+	ignored := pcap.Close() // close any existing pcap sink
+
+	if fpcap < 0 {
+		log.I("netstack: pcap closed (ignored-err? %v)", ignored)
+		return nil // nothing else to do; pcap is closed
+	} else if fpcap < 3 {
+		// if fpcap is 0, 1, or 2 then pcap is written to stdout
+		ok := pcap.log(true)
+		log.I("netstack: pcap(int32 %d)/log(%t)", fpcap, ok)
+		return nil // fdbased will write to stdout
+	} else {
+		fout := os.NewFile(uintptr(fpcap), "")
```

Comment:
Does this open the file in *append* mode?

Reply:
`os.Newfile()` doesn't open the file, it only return a `*File`, the file mode is able to be specified by the kotlin layer.
e.g.
`xxresolver.openFileDescriptor(uri, "wa")`
https://developer.android.com/reference/kotlin/android/content/ContentResolver#openFileDescriptor(android.net.Uri,%20kotlin.String)